src/main/java/io/cryostat/platform/ServiceRef.java

```diff
@@ -63,18 +69,111 @@ public URI getServiceUri() {
         return Optional.ofNullable(alias);
     }
 
+    public void setLabels(Map<String, String> labels) {
+        this.labels.clear();
+        this.labels.putAll(labels);
+    }
+
+    public Map<String, String> getLabels() {
+        return Collections.unmodifiableMap(labels);
+    }
+
+    public void setPlatformAnnotations(Map<String, String> annotations) {
+        this.annotations.platform.clear();
+        this.annotations.platform.putAll(annotations);
+    }
```

Comment:
Is this implemented yet? I assume this would contain Kubernetes annotations?

Reply:
Yes exactly, but no it is not implemented yet for Kubernetes (or any platform). The `labels` field of the ServiceRef is also intended to contain labels provided by the platform but is not yet implemented anywhere.

Reply:
For context, the reason it isn't implemented anywhere is because in this PR, nothing in Cryostat knows about Pods, or any other Kubernetes resource than Endpoints really. Once this is merged and #492 is rebased on top, then we'll have a pretty clean way to integrate these two things together and apply the labels/annotations from the target Pod to these ServiceRef fields.